src/req-wrap.h

```diff
@@ -23,7 +23,9 @@ class ReqWrap : public AsyncWrap {
  private:
   friend class Environment;
   ListNode<ReqWrap> req_wrap_queue_;
-  T req_;
+
+ protected:
+  T req_;  // Must be last.
```

Comment:
I'd like to add some more information about the reason for this requirement, that it be the last member. Moving the protected member to be before the private members will cause failures in tests, but I've not been able to find the reason for this yet. @bnoordhuis Are there any resources (links, docs) you could point me to, to help me understand this?